src/components/Items/ItemEditor.vue

```diff
@@ -0,0 +1,173 @@
+      ></ItemDropDown>
+
+      <!-- previous item button -->
+      <icon-button
+        class="rounded-tl-xl rounded-bl-xl"
+        :iconConfig="previousItemIconConfig"
+        @click="updateSelectedItemIndex(selectedItemIndex - 1)"
+        :class="{ 'opacity-50': isFirstItem }"
+        :disabled="isFirstItem"
+      ></icon-button>
+
+      <!-- next item button -->
+      <icon-button
+        class="rounded-tr-xl rounded-br-xl"
+        :iconConfig="nextItemIconConfig"
+        @click="updateSelectedItemIndex(selectedItemIndex + 1)"
+        :class="{ 'opacity-50': isLastItem }"
+        :disabled="isLastItem"
+      ></icon-button>
+
+      <!-- add item button -->
+      <icon-button class="rounded-xl" :iconConfig="addItemIconConfig"></icon-button>
+
+      <!-- delete item button -->
+      <icon-button
+        class="rounded-xl bg-delete-button"
+        :iconConfig="deleteItemIconConfig"
+      ></icon-button>
+    </div>
+
+    <!-- item editor -->
+    <div class="h-full border-2 rounded-t-xl mr-2 ml-2 p-2 item-editor-box">
+      <!-- question input box -->
+      <input-text
+        :placeholder="'placeholder'"
+        :title="'Question'"
+        v-model:value="questionText"
+        ref="questionText"
+      ></input-text>
+    </div>
+  </div>
+</template>
+
+<script>
+import IconButton from "../UI/Buttons/IconButton.vue";
+import ItemDropDown from "../UI/DropDownMenu/ItemDropDown.vue";
+import InputText from "../UI/Text/InputText.vue";
+
+export default {
+  name: "ItemEditor",
+
+  data() {
+    return {
+      previousItemIconConfig: {
+        enabled: true,
+        iconName: "chevron-left-solid",
+      },
+      nextItemIconConfig: {
+        enabled: true,
+        iconName: "chevron-right-solid",
+      },
+      addItemIconConfig: {
+        enabled: true,
+        iconName: "plus-solid",
+      },
+      deleteItemIconConfig: {
+        enabled: true,
+        iconName: "delete",
+      },
+      selectedItemIndex: 0,
+    };
+  },
+
+  props: {
+    itemList: {
+      default: () => [],
+      type: Array,
+    },
+  },
+
+  components: {
+    ItemDropDown,
+    IconButton,
+    InputText,
+  },
+
+  created() {},
+
+  methods: {
+    capitalizeFirstLetter(str) {
+      return str.charAt(0).toUpperCase() + str.slice(1);
+    },
+    updateSelectedItemIndex(index) {
+      this.selectedItemIndex = index;
+    },
+  },
+
+  computed: {
+    localItemList: {
+      // a local copy of the item list
+      get() {
+        return this.itemList;
+      },
+      set(localItemList) {
+        this.$emit("update:itemList", localItemList);
+      },
+    },
+    itemOptionsList() {
+      // preparing an options list to pass to the dropdown
+      // TODO: ditch the primevue implementation
+      var optionsList = [];
+      this.localItemList.forEach((item, itemIndex) => {
+        var currentItem = {};
+        currentItem["index"] = itemIndex;
+        var itemType = this.capitalizeFirstLetter(item.type);
```

Comment:
I don't think we should control this capitalization here - maybe do some validation while it is selected outside this itself. Can skip for now - as all are question types only - but I don't think this should be done here.

Reply:
As we discussed, I'm keeping this now. Is that cool right?